### PR TITLE
Nfdiv 1163 solicitor confirm service page

### PR DESCRIPTION
**Pull request by mark-butler-solirius**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/NFDIV-1163

### Change description ###
Implement the Solicitor Confirm Service page with transition from AwaitingService state to AoSWaiting

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorConfirmService.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            .grant(READ,
+                CASE_WORKER,
+                SUPER_USER,
+                LEGAL_ADVISOR));
+    }
+
+    private List<String> validateConfirmService(Application application) {
+        List<String> errors = new ArrayList<>();
+
+        if (!application.getSolServiceMethod().equals(ServiceMethod.SOLICITOR_SERVICE)) {
```
Comment (mapingo) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorConfirmService.java:103]: There is an Application method that checks for solicitorService method, isSolicitorServiceMethod

File: src/main/java/uk/gov/hmcts/divorce/solicitor/service/task/SetConfirmServiceDueDate.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.task.CaseTask;
+
+import java.time.Clock;
+import java.time.LocalDate;
+
+@Component
+@Slf4j
+public class SetConfirmServiceDueDate implements CaseTask {
```
Comment (mapingo) [inline src/main/java/uk/gov/hmcts/divorce/solicitor/service/task/SetConfirmServiceDueDate.java:16]: Were you going to add any unit tests for the SolicitorSubmitConfirmService and SetConfirmServiceDueDate?

Comment (mark-butler-solirius) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/service/task/SetConfirmServiceDueDate.java:16]: Good point - I will add them now

File: src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorConfirmService.java (hunk 1 above)
Comment (mark-butler-solirius) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorConfirmService.java:103]: Ah, will update

File: src/main/java/uk/gov/hmcts/divorce/solicitor/service/task/SetConfirmServiceDueDate.java (hunk 2 above)
Comment (mark-butler-solirius) [reply src/main/java/uk/gov/hmcts/divorce/solicitor/service/task/SetConfirmServiceDueDate.java:16]: Tests added - PR branch rebuilding....

**Review by mapingo — APPROVED** [review]

**Pull request merged**